src/cargo/sources/registry/remote.rs

```diff
@@ -301,13 +307,12 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
     }
 
     fn invalidate_cache(&mut self) {
-        if !self.updated {
-            self.needs_update = true;
-        }
+        // To fully invalidate, undo `mark_updated`s work
+        self.needs_update = true;
```

Comment:
@arlosi thoughts?

Reply:
I believe the intent of `updated_sources` was to prevent fetching the repo more than once per session for scenarios like `publish` (where we do an update for fetching `config.json` as well as a later update for the verification build).

Maybe it's not a big deal to do an extra fetch here though, since we'll be doing many more fetches with the polling for publish anyway and it would simplify things.

Reply:
Thinking some more about this, I feel like more investigation is needed into `updated_sources` before clearing it here in case there are areas that are relying on this behavior.  I feel like that effort is best decoupled from the current effort.

Reply:
Sound reasonable. We can deal with it afterward.